cert-manager-multi-arch.md

```diff
@@ -0,0 +1,728 @@
+
+Here we'll create a SelfSigned `ClusterIssuer`, issue a root `Certificate` and then use that root as a CA issuer.
+
+```yaml
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: selfsigned-cluster-issuer #----(1)
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: selfsigned-root-ca
+  namespace: cert-manager
+spec:
+  isCA: true
+  commonName: selfsigned-root-ca # Note this
+  secretName: root-ca-key-pair #----(2)
+  privateKey:
+    algorithm: ECDSA
+    size: 256
+  issuerRef:
+    name: selfsigned-cluster-issuer # match (1)
+    kind: ClusterIssuer
+    group: cert-manager.io
+---
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: ca-issuer
+spec:
+  ca:
+    secretName: root-ca-key-pair # match (2)
+```
+
+Feel free to modify the resource names, and take note of the `commonName` for verification purposes. The root ca will be stored in the `root-ca-key-pair` secret within the `cert-manager` namespace by cert-manager operator.
+
+### Issue Certificate
+
+Cert-manager's `Certificate` resource facilitates automatic issuance, renewal, and storage of signed certificates, contingent on configured `Issuer` or `ClusterIssuer` resources. So, we will next create a `Certificate` using the above created CA `ClusterIssuer`, for the default ingress controller in `openshift-ingress` namespace.
+
+*Note:* `dnsNames`, `commonName` should be `"*.apps.<cluster-domain>"` for the certificate. For example : `"*.apps.sandbox-cluster.mydomain.example.com"`.
+
+You can use `oc whoami --show-server` command to get the correct cluster-domain.
+
+```yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: ingress-wildcard-cert
+  namespace: openshift-ingress
+spec:
+  isCA: false
+  commonName: "*.apps.<cluster-domain>" # change me
+  dnsNames:
+    - "apps.<cluster-domain>"           # change me
+    - "*.apps.<cluster-domain>"         # change me
+  usages:
+    - server auth
+  issuerRef:
+    kind: ClusterIssuer
+    name: ca-issuer
+  secretName: ingress-wildcard-tls
+```
+The relevant certificate will be created and stored in the `ingress-wildcard-tls` secret and this is the certificate which we all care about. :+1: 
+
+### Change ingress controller's default certificate
+
+Next, we'll update the `defaultCertificate` reference in the default `IngressController` CR by patching it to point to the new certificate secret.
+
+```shell
+$ oc patch ingresscontroller.operator default \
+--type=merge -p \
+'{"spec":{"defaultCertificate": {"name": "ingress-wildcard-tls"}}}' \
+-n openshift-ingress-operator
+```
+
+
+Once the certificate is replaced, all applications, the web console and CLI, will be secured with the specified certificate for encryption.
+
+### Replace the CA bundle certificate
```

Comment:
This step is optional, right? You can mention this sub-section as Optional, if it absolutely is non-necessary.

Reply:
Checking it out further, and the use case you want to add makes me feel this is essential and well non-optional. Then we should probably mention that this step is only necessary when using custom CA to manage the trusted certs inside OpenShift and that Internet CAs like the certs issued online can skip this step altogether.